content/tutorials/spatial_transform_crs_2/index.Rmd

```diff
@@ -0,0 +1,495 @@
+}
+```
+
+The code assumes that you have a `gisclubdata_path` object defined (directory path as a string).
+
+## Joining and mapping polygon layers
+
+For a given `municipalities` and `watercourses` dataset, we would like to append the municipality name to the watercourses by making a spatial join.
+
+### Reading data and trying to join
+
+Let's read the data file of Belgian municipalities:
+
+```{r}
+path_municipalities <- file.path(gisclubdata_path, "gemeenten_belgie.shp")
+municipalities <- read_sf(path_municipalities)
+```
+
+It looks like this:
+
+```{r paged.print=FALSE}
+municipalities
+```
+
+It appears that the points have three coordinate dimensions (`XYZ`).
+However the features have only two useful dimensions:
+
+```{r}
+st_dimension(municipalities)
+```
+
+Some tools -- e.g. spherical geoprocessing -- expect XY data.
+So we will drop the Z dimension with `st_zm()`.
+Meanwhile we select the only necessary attribute and rename it, using `select()`.
+
+```{r}
+municipalities <- 
+  municipalities %>% 
+  select(municipality = T_MUN_NL) %>% 
+  st_zm()
+```
+
+We perform similar operations on the `watercourses` dataset.
+Here, data are already defined as two dimensions.
+
+```{r}
+path_watercourses <- file.path(gisclubdata_path, "watergangen_antw.shp")
+watercourses <- read_sf(path_watercourses)
+```
+
+```{r paged.print=FALSE}
+watercourses
+```
+
+Note that not all columns are printed.
+Let's preview all columns with `glimpse()`:
+
+```{r}
+glimpse(watercourses)
+```
+
+We only select some:
+
+```{r}
+watercourses <- 
+  watercourses %>% 
+  select(polygon_id = OIDN,
+         vhag_code = VHAG,
+         watercourse = NAAM)
+```
+
+```{r paged.print=FALSE}
+watercourses
+```
+
+In the above prints of the `municipalities` and `watercourses` objects, you could already see their coordinate reference system (CRS).
+With `st_crs()` you can extract the `crs` object; it is printed as:
+
+```{r}
+st_crs(municipalities)
+```
+
+The CRS name '`r st_crs(municipalities)$Name`' can also be returned by `st_crs(municipalities)$Name`.
+As seen from the WKT string, this is a geographical CRS.
+
+However for `watercourses` the CRS is missing:
+
+```{r}
+st_crs(watercourses)
+```
+
+Can we make the spatial join already?
+
+```{r}
+st_join(watercourses, municipalities) %>% try
+```
+
+Fortunately, this doesn't work: `st_join()` requires its inputs to belong to the same CRS.
+
+Well, we know that `watercourses` is in the projected CRS `EPSG:31370` (`r st_crs(31370)$Name`), so we can just update it:
+
+```{r}
+st_crs(watercourses) <- "EPSG:31370"
+```
+
+Since the CRSs are still different, this should *not* be sufficient for the join.
+
+```{r}
+st_join(watercourses, municipalities) %>% try
+```
+
+Indeed!
+
+Let's transform `municipalities` to `EPSG:31370`:
+
+```{r}
+municipalities_31370 <- st_transform(municipalities, "EPSG:31370")
+```
+
+And now...
+
+```{r paged.print=FALSE}
+(watercourses_comm <- st_join(watercourses, municipalities_31370))
```

Comment:
Maybe explain that `st_join` default is based on `st_intersects` (probably more rows in the result than rows in watercourses)?

Reply:
> probably more rows in the result than rows in watercourses

Indeed.

Good suggestion!

Reply:
Done in 7347e0ea